scripts/pylib/pytest-twister-harness/src/twister_harness/device/utils.py

```diff
@@ -41,11 +41,16 @@ def terminate_process(proc: subprocess.Popen) -> None:
     """
     Try to terminate provided process and all its subprocesses recursively.
     """
-    for child in psutil.Process(proc.pid).children(recursive=True):
-        try:
-            os.kill(child.pid, signal.SIGTERM)
-        except (ProcessLookupError, psutil.NoSuchProcess):
-            pass
+
+    try:
+        for child in psutil.Process(proc.pid).children(recursive=True):
+            try:
+                os.kill(child.pid, signal.SIGTERM)
+            except (ProcessLookupError, psutil.NoSuchProcess):
+                pass
+    except (ProcessLookupError, psutil.NoSuchProcess):
+        pass
```

Comment:
The code will work but it's good to keep it relevant. One exception is targeting `os` and the other `psutil`. A relevant error should correspond to each of them (i.e. os won't raise psutill exception and vice versa) hence my suggestion:

```suggestion
    try:
        for child in psutil.Process(proc.pid).children(recursive=True):
            try:
                os.kill(child.pid, signal.SIGTERM)
            except (ProcessLookupError):
                pass
    except (psutil.NoSuchProcess):
        pass
```

Or in this case if we just want to pass over exception a `suppress` can be used:

```
    from contextlib import suppress
    
    with suppress(psutil.NoSuchProcess):            
        for child in psutil.Process(proc.pid).children(recursive=True):
            with suppress(ProcessLookupError):
                os.kill(child.pid, signal.SIGTERM)
```

Reply:
that's not quite right. both `Process()` and `kill()` can raise both of these exceptions.